docs/develop/create-addin-commands.md

```diff
@@ -146,55 +96,50 @@ The following is an example of the **\<FunctionFile\>** element.
-
-The following code shows how to implement the function used by **\<FunctionName\>**.
-
-```html
-<script>
-    // The initialize function must be run each time a new page is loaded.
-    (function () {
-        Office.initialize = function (reason) {
-            // If you need to initialize something you can do so here.
-        };
-    })();
-
-    // Define the function.
-    function writeText(event) {
-
-        // Implement your custom code here. The following code is a simple example.  
-        Office.context.document.setSelectedDataAsync("Function command works. Button ID=" + event.source.id,
-            function (asyncResult) {
-                const error = asyncResult.error;
-                if (asyncResult.status === Office.AsyncResultStatus.Failed) {
-                    // Show error message.
-                }
-                else {
-                    // Show success message.
-                }
-            });
-
-        // Calling event.completed is required. event.completed lets the platform know that processing has completed.
-        event.completed();
-    }
-    
-    // You must register the function with the following line.
-    Office.actions.associate("writeText", writeText);
-</script>
+> Office.js must be initialized before the add-in command logic runs. See [Initialize your Office Add-in](initialize-add-in.md) for more information.
+
+The following code shows an example function used by **\<FunctionName\>**. Note the call to **event.completed**, This signals that you have successfully handled the event. When a function is called multiple times, such as multiple clicks on the same add-in command, all events are automatically queued. The first event runs automatically, while the other events remain on the queue. When your function calls **event.completed**, the next queued call to that function runs. You must implement **event.completed**, otherwise your function will not run.
+
+```js
+// Initialize the Office Add-in.
+Office.onReady(() => {
+  // If needed, Office.js is ready to be called
+});
+
+// The command function.
+async function highlightSelection(event) {
+
+    // Implement your custom code here. The following code is a simple Excel example.  
+    try {
+          await Excel.run(async (context) => {
+              const range = context.workbook.getSelectedRange();
+              range.format.fill.color = "yellow";
+              await context.sync();
+          });
+      } catch (error) {
+          // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
+          console.error(error);
+      }
+
+    // Calling event.completed is required. event.completed lets the platform know that processing has completed.
+    event.completed();
+}
+
+// You must register the function with the following line.
+Office.actions.associate("highlightSelection", highlightSelection);
+
 ```
```

Comment:
Should this sample code be moved to 'Button controls' in Step 5 since the \<FunctionName> element is referenced there? If it's moved, consider changing the \<FunctionName> element value of that section's manifest sample to `highlightSelection`.

Reply:
Good call.